src/components/structures/MessagePanel.js

```diff
@@ -238,26 +239,41 @@ module.exports = React.createClass({
         return !this._isMounted;
     },
 
-    _getEventTiles: function() {
+    // TODO: Implement granular (per-room) hide options
+    _shouldShowEvent: function(mxEv) {
+        // console.log("_shouldShowEvent ", mxEv.getId(), mxEv);
         const EventTile = sdk.getComponent('rooms.EventTile');
+        if (!EventTile.haveTileForEvent(mxEv)) {
+            return false; // no tile = no show
+        }
+
+        // Always show highlighted event
+        if (this.props.highlightedEventId === mxEv.getId()) return true;
+
+        return !shouldHideEvent(mxEv, this._syncedSettings);
+    },
+
+    _getEventTiles: function() {
         const DateSeparator = sdk.getComponent('messages.DateSeparator');
         const MemberEventListSummary = sdk.getComponent('views.elements.MemberEventListSummary');
 
+        let visible = this.props.readMarkerVisible;
+
         this.eventNodes = {};
 
         var i;
 
         // first figure out which is the last event in the list which we're
         // actually going to show; this allows us to behave slightly
-        // differently for the last event in the list.
+        // differently for the last event in the list. (eg show timestamps)
```

Comment:
"show timestamps" is the var name that eventually gets passed through the to event component. I really don't care though.